crates/sui-core/src/authority/authority_store.rs

```diff
@@ -1136,6 +1136,16 @@ impl<const A: bool, const B: bool, S: Eq + Serialize + for<'de> Deserialize<'de>
     }
 }
 
+/// A wrapper to make Orphan Rule happy
+pub struct ArcWrapper<T>(pub Arc<T>);
+
+impl ModuleResolver for ArcWrapper<AuthorityStore> {
+    type Error = SuiError;
+    fn get_module(&self, module_id: &ModuleId) -> Result<Option<Vec<u8>>, Self::Error> {
+        self.0.get_module(module_id)
+    }
+}
+
```

Comment:
Traditionally in a [NewType pattern](https://rust-unofficial.github.io/patterns/patterns/behavioural/newtype.html) you'd go with something like 
```rust
pub struct AuthorityStoreWrapper(Arc<SuiDataStore<false, true, AuthoritySignInfo>>);
impl ModuleResolver for AuthorityStoreWrapper {
...
```

In other terms, no need to host a generic newtype.

Reply:
I'm fine with either approach. Will remove generic if no one objects